docs/remote-feature-flags.md

```diff
@@ -0,0 +1,171 @@
+# Remote Feature Flags
+
+## Overview
+
+This document outlines the process for adding and managing remote feature flags in MetaMask (Extension and Mobile). Remote feature flags allow us to control feature availability and behavior in production without requiring a new release. They are created on the Launch Darkly platform and distributed by our internal API [ClientConfigAPI](https://github.com/consensys-vertical-apps/mmwp-client-config-api). Please read the [Remote feature flags ADR](https://github.com/MetaMask/decisions/pull/43) for more details.
+
+## Implementation Guide
+
+### 1. Creating a New Feature Flag
+
+1. Name your flag with team prefix (e.g., `confirmations-blockaid`, `ramps-card`)
+2. Request LaunchDarkly access from TechOps. Here's a template email:
+
+```
+Subject: Requesting access to LaunchDarkly for [team]
+
+Dear TechOps,
+Can I please get access to LaunchDarkly with the "Metamask Extension and Mobile" role? (role key: `metamask-config`)
+
+Thanks!
+```
+
+3. Select appropriate project in LaunchDarkly:
+   - Extension: "MetaMask Client Config API - Extension"
+   - Mobile: "MetaMask Client Config API - Mobile"
+4. Create the flag following the [LaunchDarkly flag creation guide](https://docs.launchdarkly.com/home/flags/new)
+
+### 2. Using Remote Feature Flags in MetaMask
+
+#### Extension
+
+To use a remote feature flag in the MetaMask extension, follow these steps:
+
+1. Add selector from `import { getRemoteFeatureFlags } from 'path/to/selectors';` in your component file.
+2. Use the selector to get the feature flag value.
+
+```typescript
+const { testFeatureFlag } = getRemoteFeatureFlags(state);
+```
+
+3. Use the feature flag value in your component.
+
+```typescript
+if (testFeatureFlag) {
+  // Use the feature flag value
+}
+```
+
+#### Mobile
+
+##### Create a selector for your feature flag
+
+- Location: `app/selectors/featureFlagsController`
+- Create a new folder with your feature flag name
+- Follow the structure from `app/selectors/featureFlagsController/minimumAppVersion`
+
+Your selector must include:
+
+- State selectors for each feature flag value
+- Fallback values for invalid remote flag values
+- TypeScript types
+- Unit tests
+- Any required business logic for value manipulation
+
+> [!IMPORTANT]
+> Always use the specific feature flag selector when accessing values. Accessing feature flags directly from Redux state or the main selector bypasses fallback values and can cause crashes.
+
+### 3. Remote Feature Flags with thresholds
```

Comment:
I'd rename this part `3. Creating feature flag in LaunchDarkly`.

I'd also swap this part with `2. Using Remote Feature Flags in MetaMask` as we first need to create the flag in LaunchDarkly, then implement it in the client.

Reply:
Actually the creation part is the first step in `implementation guide`. Please check again and see if anything is missing 🙏🏻

Reply:
Sounds good!